doobie-hikari/src/main/scala/com/avast/sst/doobie/DoobieHikariModule.scala

```diff
@@ -0,0 +1,45 @@
+package com.avast.sst.doobie
+
+import cats.effect.{Async, Blocker, ContextShift, Resource, Sync}
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.metrics.MetricsTrackerFactory
+import doobie.hikari.HikariTransactor
+
+import scala.concurrent.ExecutionContext
+
+object DoobieHikariModule {
+
+  /** Makes [[doobie.hikari.HikariTransactor]] initialized with the given config.
+    *
+    * @param boundedConnectExecutionContext [[scala.concurrent.ExecutionContext]] used for creating connections (should be bounded!)
+    */
+  def make[F[_]: Async](
+    config: DoobieHikariConfig,
+    boundedConnectExecutionContext: ExecutionContext,
+    blocker: Blocker,
+    metricsTrackerFactory: Option[MetricsTrackerFactory] = None
+  )(implicit cs: ContextShift[F]): Resource[F, HikariTransactor[F]] = {
+    for {
+      hikariConfig <- Resource.liftF(Sync[F].delay(makeHikariConfig(config, metricsTrackerFactory)))
+      transactor <- HikariTransactor.fromHikariConfig(hikariConfig, boundedConnectExecutionContext, blocker)
+    } yield transactor
+  }
+
+  private def makeHikariConfig(config: DoobieHikariConfig, metricsTrackerFactory: Option[MetricsTrackerFactory]) = {
```

Comment:
maybe to be super purist, since we want to assume this could have sideeffects, maybe the return type should be `F[HikariConfig]` :thinking: 

I know the method is private -- but it already confused me a bit -- it's good to use types even for documentation and explaining intent

Reply:
https://github.com/openbouquet/HikariCP/blob/master/src/main/java/com/zaxxer/hikari/HikariConfig.java#L113

it truly is side-effectful